python/cudf/cudf/comm/serialize.py

```diff
@@ -23,22 +23,29 @@
     def cuda_serialize_cudf_object(x):
         with log_errors():
             header, frames = x.serialize()
-            assert all(isinstance(f, cudf.core.buffer.Buffer) for f in frames)
+            assert all((type(f) is cudf.core.buffer.Buffer) for f in frames)
             return header, frames
 
     # all (de-)serializtion are attached to cudf Objects:
     # Series/DataFrame/Index/Column/Buffer/etc
     @dask_serialize.register(serializable_classes)
     def dask_serialize_cudf_object(x):
+        header, frames = cuda_serialize_cudf_object(x)
         with log_errors():
-            header, frames = x.serialize()
             frames = [f.to_host_array().data for f in frames]
             return header, frames
 
     @cuda_deserialize.register(serializable_classes)
     @dask_deserialize.register(serializable_classes)
     def deserialize_cudf_object(header, frames):
         with log_errors():
+            if header["serializer"] == "cuda":
+                assert all(
+                    hasattr(f, "__cuda_array_interface__") for f in frames
+                )
+            if header["serializer"] == "dask":
```

Comment:
Should have been `elif`. Fixing in PR ( https://github.com/rapidsai/cudf/pull/4537 ).

Reply:
How can the key `serializer` have multiple values?

Reply:
Often in dask a list of possible serializers is passed:
> serializers=("cuda", "dask", "pickle", "error")

if one fails the next one is used

Reply:
Yeah though the `header` should only contain one at the top-level. IOW this shouldn't be a collection.

Reply:
This thread continues here: https://github.com/rapidsai/cudf/pull/4537#pullrequestreview-375722736